docs/UsingLayering.md

```diff
@@ -0,0 +1,165 @@
+
+Nothing will stop you at this point from using a completely arbitrary image, but if you want to succeed, you should derive your layered image from the base image matching your cluster.
+
+> NOTE: at some point we intend these images to be published to a predictable place for you to easily retrieve, but for now you will want to pull the image out of the release payload matching your cluster
+
+#### On an existing cluster
+
+```bash
+oc adm release info --image-for rhel-coreos-8
+```
+
+#### Or before you build your cluster
+
+```bash
+oc adm release info --image-for rhel-coreos-8 quay.io/openshift-release-dev/ocp-release:your_release_here
+```
+
+### 2. "Layer" Some Content On Top Of It
+
+#### Generic Examples
+
+There are a lot of things you can do here, I suggest using the [Layering Examples](https://github.com/coreos/coreos-layering-examples) as a guideline for now, but they were written for Fedora, and obviously you'll need to use RPM packages that match your OS and version (RHCOS, SCOS, etc).
+
+#### Entitled Packages
+
+For RHCOS, using an entitled RHEL host is the easy way to go, podman should pass the entitlements through and it should just work:
+
+```dockerfile
+FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:69cf9251c36c4df35cd676657c7fc1cb7706a4dccdf5dab4743115e2cac48f3b
+RUN rpm-ostree install libreswan && \
+    rpm-ostree cleanup -m && \
+    ostree container commit && rm /etc/yum.repos.d/redhat.repo
+```
+
+> WARNING: take note of the removal of the redhat.repo file. Its presence is a side-effect of the entitled build. If you leave it there, rpm-ostree will try to use it later and fail because the certificates that enable its use will no longer be present after the build.
+
+> NOTE: For now, the standard `rpm-ostree` rules apply, so if you install packages that do things like [install into /opt/](https://github.com/coreos/rpm-ostree/issues/233)  you might not be completely successful.
+
+> NOTE: `MachineConfig` takes precedence over config files included in a derived image. Conflicts will not break anything, but right now `MachineConfig` always wins and will just overwrite the change.
+
+#### "I Just Want To See It Work"
+
+We recommend against using CentOS stream packages in lieu of entitled packages for RHCOS, but if you're just playing with it and want to see it work, you *could* do something totally unsupported like this:
+
+```dockerfile
+FROM quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:ffa3568233298408421ff7da60e5c594fb63b2551c6ab53843eb51c8cf6838ba
+RUN rpm-ostree install http://mirror.centos.org/centos/8-stream/AppStream/x86_64/os/Packages/ltrace-0.7.91-28.el8.x86_64.rpm && rpm-ostree cleanup -m
+```
+
+> WARNING: Take care that you either run your cleanup as part of your installation command so the temp files/caches don't end up in your final image. If you need to do it in multiple steps, you can run your podman build with `--squash` to prevent the inclusion of the intermediate layers.
```

Comment:
minor nit, `Take care that you either run your ...`, did you mean to have an "or" at some point in the sentence?

Reply:
Yes, that was originally "either run your cleanup or squash" and I rearranged it. Excellent catch, thank you.